### PR TITLE
renaming: FCM is the new version of GCM

### PR DESCRIPTION
**Pull request by matzew**

### CODE REVIEW

File: impl/src/main/java/org/jboss/aerogear/proxy/command/FCMProxyCommand.java [hunk 1]
```diff
@@ -17,53 +17,53 @@
...
         validate();
 
         startNotificationRegisterEndpoint(notificationEndpointHost, notificationEndpointPort);
 
-        MockingGCMServerBackgroundThread backgroundThread = new MockingGCMServerBackgroundThread(gcmMockServerHost,
-            gcmMockServerPort,
-            new File(gcmCertificate),
-            new File(gcmCertificateKey));
+        MockingFCMServerBackgroundThread backgroundThread = new MockingFCMServerBackgroundThread(gcmMockServerHost,
+                fcmMockServerPort,
```
Comment (josemigallas) [inline impl/src/main/java/org/jboss/aerogear/proxy/command/FCMProxyCommand.java:52]: There is one more tab here

File: impl/src/main/java/org/jboss/aerogear/proxy/command/FCMProxyCommand.java [hunk 2]
```diff
@@ -17,53 +17,53 @@
...
 
     @Option(name = "--httpProxyHost", description = "defaults to 127.0.0.1")
     private String httpProxyHost = "127.0.0.1";
 
     @Option(name = "--httpProxyPort", description = "defaults to 16000")
     private int httpProxyPort = 16000;
 
-    @Option(name = "--gcmMockServerHost", description = "defaults to 127.0.0.1")
+    @Option(name = "--fcmMockServerHost", description = "defaults to 127.0.0.1")
     private String gcmMockServerHost = "127.0.0.1";
```
Comment (josemigallas) [inline impl/src/main/java/org/jboss/aerogear/proxy/command/FCMProxyCommand.java]: What about this? gcmMockServerHost -> fcmMockServerHost

File: impl/src/main/java/org/jboss/aerogear/proxy/fcm/FCMNotificationRegister.java [hunk 3]
```diff
@@ -1,23 +1,23 @@
...
 
-    public static synchronized void addNotification(GCMNotification notification) {
+    public static synchronized void addNotification(FCMNotification notification) {
         notifications.add(notification);
     }
 
-    public static synchronized List<GCMNotification> getNotifications() {
+    public static synchronized List<FCMNotification> getNotifications() {
         return notifications;
     }
```
Comment (josemigallas) [inline impl/src/main/java/org/jboss/aerogear/proxy/fcm/FCMNotificationRegister.java:23]: I cannot comment there but in line 25 there's also a GCM reference

File: impl/src/main/java/org/jboss/aerogear/proxy/fcm/MockingFCMServerBackgroundThread.java [hunk 4]
```diff
@@ -13,7 +13,7 @@
 
 import javax.net.ssl.SSLException;
 
-public class MockingGCMServerBackgroundThread extends Thread {
+public class MockingFCMServerBackgroundThread extends Thread {
 
     private final String gcmMockServerHost;
```
Comment (josemigallas) [inline impl/src/main/java/org/jboss/aerogear/proxy/fcm/MockingFCMServerBackgroundThread.java]: What about all these fields?

File: model/src/main/java/org/jboss/aerogear/proxy/endpoint/model/NotificationRegisterResponseHelper.java [hunk 5]
```diff
@@ -18,7 +17,7 @@
             deviceTokens.add(apnsNotification.getDeviceToken());
         }
 
-        for (final GCMNotification gcmNotification : notificationRegisterResponse.getGcmNotifications()) {
+        for (final FCMNotification gcmNotification : notificationRegisterResponse.getFcmNotifications()) {
```
Comment (josemigallas) [inline model/src/main/java/org/jboss/aerogear/proxy/endpoint/model/NotificationRegisterResponseHelper.java]: Class and method are changes, however not the variable.

File: model/src/test/java/org/jboss/aerogear/proxy/endpoint/model/ModelSerializationTestCase.java [hunk 6]
```diff
@@ -26,7 +26,7 @@ public void apnsNotificationSerializationTest() {
 
     @Test
     public void gcmNotificationSerializationTest() {
```
Comment (josemigallas) [inline model/src/test/java/org/jboss/aerogear/proxy/endpoint/model/ModelSerializationTestCase.java]: Test name is still GCM

**Review by josemigallas — CHANGES_REQUESTED** [review]
There are other files like pom and readme that still have references to GCM

**Comment by matzew** [comment]
@josemigallas updated

**Comment by josemigallas** [comment]
Awesome :+1:

**Review by josemigallas — APPROVED** [review]

**Pull request merged**